[FEAT] 일정 화면 UI 구현

paengjiwoo opened the pull request:
## ✅ 이슈 번호
#14 

## ✅ 작업 내용
- HomePage.tsx 화면 구현
  - 일간, 월간 이동할 Tabs 컴포넌트 구현
  - 일간 페이지 일정 리스트 구현
- 일정 상세 페이지 ScheduleDetailPage.tsx UI 구현

## ✅ 공유 사항
- PrevIcon path의 className을 'fill-current'로 변경했습니다.
- common 폴더에 CategoryTag 컴포넌트를 구현했습니다. Prop으로`label="카테고리 이름"`을 보내면 자동으로 색상이 맵핑됩니다.

paengjiwoo committed:
feat: homePage 구조 생성 및 Tabs 컴포넌트 생성

- schedule components 생성
  - CalendarView, EventContainer, EventItem, Tabs
- Tabs 컴포넌트 기능 연결 및 스타일

paengjiwoo committed:
feat: 일간 View Item Card 생성 및 Category Tag 생성

- EventItem Card 구현
- CategoryTag 생성
  - category enum 모델 정의

paengjiwoo committed:
chore: category.model.ts -> category.ts 파일명 변경

paengjiwoo committed:
feat: PrevIcon Prop color 추가

- 기본 색상을 #ffffff 로 설정
- color 인자에 hex color code 할당할 경우, 색상 변경 가능

paengjiwoo committed:
feat: 홈 화면 일정 item card 및 일정 상세 페이지 Link 연결

paengjiwoo committed:
feat: 일정 상세 페이지 content 생성

- InfoItem 컴포넌트로 content간 스타일 일관성 유지

paengjiwoo committed:
chore: PrevIcon path fill -> className 변경

- fill="hex code" 대신 className="fill-current" 작성

paengjiwoo committed:
feat: twMerge 사용하도록 cn 추가

jellysoo97 commented on src/components/common/CategoryTag.tsx:
```diff
@@ -0,0 +1,23 @@
+  label: Categories
+}
+
+const CategoryTag = ({ label }: Props) => {
```
일정 수정에서 버튼으로도 사용할 수 있게 props를 추가해서 용도에 따라 다른 태그를 렌더링하는건 어떤가요?

paengjiwoo replied:
좀 더 확장성 있게 구성하는 것도 좋겠네요. 현재는 span 태그를 활용해 component를 구성하고 있는데 button의 형태로도 활용될 수 있도록 해보겠습니다.
아까 아이콘 컴포넌트의 경우에도 확장성 있게 구성하니 사용하기 편리하더라구요.

paengjiwoo committed:
feat: categoryTag -> span 및 button 태그 모두 활용되도록 수정

! 'as' prop에 'button'을 보내 button 태그로 활용할 수 있습니다

- React.HTMLAttributes<HTMLElement> 타입 & ...props 추가
  - HTML 속성들 prop 가능
